crates/orderbook/openapi.yml

```diff
@@ -540,32 +530,113 @@ components:
-                - $ref: "#/components/schemas/Address"
-              nullable: true
-            quoteId:
+      type: object
+      properties:
+        sellToken:
+          description: see `OrderParameters::sellToken`
+          allOf:
+            - $ref: "#/components/schemas/Address"
+        buyToken:
+          description: see `OrderParameters::buyToken`
+          allOf:
+            - $ref: "#/components/schemas/Address"
+        receiver:
+          description: see `OrderParameters::receiver`
+          allOf:
+            - $ref: "#/components/schemas/Address"
+          nullable: true
+        sellAmount:
+          description: see `OrderParameters::sellAmount`
+          allOf:
+            - $ref: "#/components/schemas/TokenAmount"
+        buyAmount:
+          description: see `OrderParameters::buyAmount`
+          allOf:
+            - $ref: "#/components/schemas/TokenAmount"
+        validTo:
+          description: see `OrderParameters::validTo`
+          type: integer
+        feeAmount:
+          description: see `OrderParameters::feeAmount`
+          allOf:
+            - $ref: "#/components/schemas/TokenAmount"
+        kind:
+          description: see `OrderParameters::kind`
+          allOf:
+            - $ref: "#/components/schemas/OrderKind"
+        partiallyFillable:
+          description: see `OrderParameters::partiallyFillable`
+          type: boolean
+        sellTokenBalance:
+          description: see `OrderParameters::sellTokenBalance`
+          allOf:
+            - $ref: "#/components/schemas/SellTokenSource"
+          default: "erc20"
+        buyTokenBalance:
+          description: see `OrderParameters::buyTokenBalance`
+          allOf:
+            - $ref: "#/components/schemas/BuyTokenDestination"
+          default: "erc20"
+        signingScheme:
+          $ref: "#/components/schemas/SigningScheme"
+        signature:
+          $ref: "#/components/schemas/Signature"
+        from:
+          description: |
+            If set, the backend enforces that this address matches what is decoded as the signer of
+            the signature. This helps catch errors with invalid signature encodings as the backend
+            might otherwise silently work with an unexpected address that for example does not have
+            any balance.
+          allOf:
+            - $ref: "#/components/schemas/Address"
+          nullable: true
+        quoteId:
+          description: |
+            Orders can optionally include a quote ID. This way the order can be linked to a quote
+            and enable providing more metadata when analyzing order slippage.
+          type: integer
+          nullable: true
+        appData:
+          description: |
+            This field comes in two forms for backward compatibility. The first form (app data hash) will eventually stop being accepted.
+          anyOf:
+            - $ref: "#/components/schemas/AppDataHash"
+            - title: Full App Data
```

Comment:
If `appDataHash` and `appData` are connected, shouldn't it be an object with the two properties, the data and the hash? (for the second type)

Reply:
The reason I did it this way is for backward compatibility on the v1 create order route.